api/lib/uploadHandler.mjs

```diff
@@ -0,0 +1,110 @@
+import multer from 'multer';
+import sanitize from 'sanitize-filename';
+import { unlink } from 'fs';
+
+import mimeTypes from '../config/mime.mjs';
+
+/**
+ * Native file handling utility.
+ */
+export default class UploadHandler {
+    sanitizedFileName = undefined;
+
+    /**
+     * Creates a new file handler.
+     * @constructor
+     * @param {string} fieldName the name of field in which the file is sent.
+     * @param {string} directory the path to store the file in.
+     * @param {string} mimeType the mime type of the file.
+     * @param {number} fileSize the maximum number of bytes allowed.
+     * @param {function} fileNameTransformer the cb for the file name.
+     */
+    constructor(
+        fieldName,
+        directory,
+        mimeType,
+        fileSize,
+        fileNameTransformer = (f) => sanitize(f.originalname),
```

Comment:
What is the plan for guidelines on what the fileName can be? What if school A creates the file `cs1.cm`  and school B also creates the file `cs1.cm`? Then are we overwriting school A's `cs1.cm` file? Do we want to add more metadata for the fileName to ensure files are not overwritten? Are we saving these files locally? What if we restart the application, then does the data persist? Are we going to move this to a database later?

Reply:
If we are just supporting CS10 for now, then this duplicate naming convention will probably not be a problem. But we should recommend some naming convention in the README or the user interface, for example `cs10_{semester}{year}.cm`

Reply:
1. Has to be less than 100 chars
2. Allowing overwrites
3. Maybe in the future
4. Stored as files in directory
5. Data persists
6. Ideally
7. naming conventions would be optimal, but it is what the user needs to input in the route, so I'm leaving it configurable to the admin

Reply:
Once we have a db we can map it to a key and call it a day